#! /bin/sh /usr/share/dpatch/dpatch-run
## 80_system_libs.dpatch by  <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make sure we won't be bitten by upstream changing libjpeg, libpng
## DP: or zlib internal version, which makes system library not used even
## DP: though --with-system-* argument is given to configure.

@DPATCH@

--- xulrunner/configure.in
+++ xulrunner/configure.in
@@ -3785,7 +3785,7 @@
     AC_CHECK_LIB(jpeg, jpeg_destroy_compress, [SYSTEM_JPEG=1 JPEG_LIBS="-ljpeg $JPEG_LIBS"], SYSTEM_JPEG=, $JPEG_LIBS)
 fi
 
-if test "$SYSTEM_JPEG" = 1; then
+if test "$SYSTEM_JPEG" = 0; then
     LIBS="$JPEG_LIBS $LIBS"
     AC_TRY_RUN( 
 	#include <stdio.h>
@@ -3830,7 +3830,7 @@
     AC_CHECK_LIB(z, gzread, [SYSTEM_ZLIB=1 ZLIB_LIBS="-lz $ZLIB_LIBS"], 
 	[SYSTEM_ZLIB= ZLIB_CFLAGS= ZLIB_LIBS=], $ZLIB_LIBS)
 fi
-if test "$SYSTEM_ZLIB" = 1; then
+if test "$SYSTEM_ZLIB" = 0; then
     LIBS="$ZLIB_LIBS $LIBS"
     AC_TRY_RUN([
     #include <stdio.h>
@@ -3883,7 +3883,7 @@
     AC_CHECK_LIB(png, png_get_valid, [SYSTEM_PNG=1 PNG_LIBS="-lpng $PNG_LIBS"],
                  SYSTEM_PNG=, $PNG_LIBS)
 fi
-if test "$SYSTEM_PNG" = 1; then
+if test "$SYSTEM_PNG" = 0; then
     LIBS="$PNG_LIBS $LIBS"
     AC_TRY_RUN(
 	#include <stdio.h>
